# Replaces https://chrome-devtools-frontend.appspot.com with trk://chrome-devtools-frontend
# This fixes a bug encountered when running domain substitution

--- a/third_party/WebKit/Source/devtools/front_end/Runtime.js
+++ b/third_party/WebKit/Source/devtools/front_end/Runtime.js
@@ -1165,7 +1165,7 @@ Runtime.experiments = new Runtime.ExperimentsSupport();
 Runtime._remoteBase = Runtime.queryParam("remoteBase");
 {(function validateRemoteBase()
 {
-    var remoteBaseRegexp = /^https:\/\/chrome-devtools-frontend\.appspot\.com\/serve_file\/@[0-9a-zA-Z]+\/?$/;
+    var remoteBaseRegexp = /^trk:\/\/chrome-devtools-frontend\/serve_file\/@[0-9a-zA-Z]+\/?$/;
     if (Runtime._remoteBase && !remoteBaseRegexp.test(Runtime._remoteBase))
         Runtime._remoteBase = null;
 })();}
--- a/third_party/WebKit/Source/devtools/front_end/devtools.js
+++ b/third_party/WebKit/Source/devtools/front_end/devtools.js
@@ -986,8 +986,8 @@ function installObjectObserve()
  */
 function sanitizeRemoteFrontendUrl()
 {
-    var remoteBaseRegexp = /^https:\/\/chrome-devtools-frontend\.appspot\.com\/serve_file\/@[0-9a-zA-Z]+\/?$/;
-    var remoteFrontendUrlRegexp = /^https:\/\/chrome-devtools-frontend\.appspot\.com\/serve_rev\/@?[0-9a-zA-Z]+\/(devtools|inspector)\.html$/;
+    var remoteBaseRegexp = /^trk:\/\/chrome-devtools-frontend\/serve_file\/@[0-9a-zA-Z]+\/?$/;
+    var remoteFrontendUrlRegexp = /^trk:\/\/chrome-devtools-frontend\/serve_rev\/@?[0-9a-zA-Z]+\/(devtools|inspector)\.html$/;
     var queryParams = location.search;
     if (!queryParams)
         return;
--- a/third_party/WebKit/Source/devtools/front_end/inspector.html
+++ b/third_party/WebKit/Source/devtools/front_end/inspector.html
@@ -7,7 +7,7 @@
 <html>
 <head>
     <meta http-equiv="content-type" content="text/html; charset=utf-8">
-    <meta http-equiv="Content-Security-Policy" content="object-src 'none'; script-src 'self' 'unsafe-eval' https://chrome-devtools-frontend.appspot.com">
+    <meta http-equiv="Content-Security-Policy" content="object-src 'none'; script-src 'self' 'unsafe-eval' trk://chrome-devtools-frontend">
     <script type="text/javascript" src="Runtime.js"></script>
     <script type="text/javascript" src="inspector.js"></script>
 </head>
--- a/chrome/browser/ui/webui/devtools_ui.cc
+++ b/chrome/browser/ui/webui/devtools_ui.cc
@@ -30,9 +30,9 @@ std::string PathWithoutParams(const std::string& path) {
       .path().substr(1);
 }
 
-const char kRemoteFrontendDomain[] = "chrome-devtools-frontend.8pp2p8t.qjz9zk";
+const char kRemoteFrontendDomain[] = "chrome-devtools-frontend";
 const char kRemoteFrontendBase[] =
-    "https://chrome-devtools-frontend.8pp2p8t.qjz9zk/";
+    "trk://chrome-devtools-frontend/";
 const char kRemoteFrontendPath[] = "serve_file";
 const char kHttpNotFound[] = "HTTP/1.1 404 Not Found\n\n";
 
